File: day1/LinkedList.js
class Node{
    constructor(value) {
        this.value = value;
        this.next = null;
    }
}

class List{
    constructor(){
        this.head = null;
        this.tail = null;
    }

    add(value){
        let ref = new Node(value);
        if (null == this.head ){
            this.head = ref;
            this.tail = ref;
        }else{
            this.tail.next = ref;
            this.tail = ref;    

        }
    }

    print(){
        for (let current = this.head; current; current = current.next){
            console.log(current.value);
        }
    }
}

let ls = new List();
ls.add(1);
ls.add(2);
ls.add(3);
ls.print();

File: day1/LinkedList.py

class Node:
    def __init__(self, value):
        self.value = value
        self.next = None

class List:
    def __init__(self):
        self.head = None
        self.tail = None

    def add(self,value):
        ref = Node(value)
        print(f"ref {ref}")
        print(f"ref.value {ref.value}")
        if self.head is None:
            self.head = ref
        else:
            self.tail.next = ref    

        self.tail = ref    

    def print(self):
        current = self.head
        while(current != None):
            print(current.value)
            current = current.next    

ls = List()
ls.add(1)
ls.add(2)
ls.add(3)
ls.print()